components-api/src/test/scala/pl/touk/nussknacker/engine/api/typed/SubclassDeterminerSpec.scala

```diff
@@ -0,0 +1,39 @@
+package pl.touk.nussknacker.engine.api.typed
+
+import org.apache.commons.lang3.ClassUtils
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+
+class SubclassDeterminerSpec extends AnyFunSuite with Matchers {
+
+  test("Should validate assignability for decimal types") {
+    StrictConversionDeterminer.isAssignable(classOf[java.lang.Long], classOf[java.lang.Integer]) shouldBe false
```

Comment:
I think we shouldn't ever check `.isAssignable` - the only things that should be public in a ConversionDeterminer is just the check taking 2 values which are types from out typing system.

Reply:
I have doubts whether being very strict with public/private here is the correct way, we could possibly end up in a situation where the next person is also surprised by how this method works. I will remove the test and add a comment in isAssignable to prevent that

Reply:
Why leave it public? Its not / shouldn't be used in this project or outside it.

Reply:
Just for testing, i agree that it shouldn't be used outside